lib/shoulda/matchers/active_record/association_matcher.rb

```diff
@@ -1394,17 +1394,23 @@ def touch_correct?
         end
 
         def class_has_foreign_key?(klass)
+          @missing = validate_foreign_key(klass)
+
+          @missing.nil?
+        end
+
+        def validate_foreign_key(klass)
           if options.key?(:foreign_key) && !foreign_key_correct?
-            @missing = foreign_key_failure_message(klass, options[:foreign_key])
-            false
-          elsif !has_column?(klass, foreign_key)
-            @missing = foreign_key_failure_message(klass, foreign_key)
-            false
-          else
-            true
+            foreign_key_failure_message(klass, options[:foreign_key])
+          elsif !has_column?(klass, to_s(actual_foreign_key))
+            foreign_key_failure_message(klass, actual_foreign_key)
           end
         end
 
+        def to_s(value)
```

Comment:
Since `to_s` is a built-in method in Ruby, maybe we should name this something else other than `to_s` so people don't get confused? Maybe `normalize_foreign_key`?

Reply:
No problem. I agree with you. Thank you for the suggestions. I will use the first one.